src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,39 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int FIELDS_MIN_LENGTH = 6;
+    private static final int USER_MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("There is no user to add");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User with same login already exists");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < FIELDS_MIN_LENGTH) {
+            throw new RegistrationException("Login should be at least " + FIELDS_MIN_LENGTH
```

Comment:
This error message correspond only to login lenght but will be thrown if login is null as well, please add specific message for null case (do for other fields as well)